### ISSUE TITLE
[MyShelf] Alter permissions of "MyShelf" collections created by users.

### ISSUE THREAD

**Issue by straleyb** [OregonDigital/OD2#1169]
### Descriptive summary

Currently the collections appear in search results, which may not be the right functionality. POSM has discussed this. @raybrarian Is the needs to suppress the MyShelf collections in the main search results?

**Comment by raybrarian** [comment]
After POSM review:
My Shelf Collections (My Shelf folders) will be set as private by default. In the My Shelf interface, there will be a button for any collection to select along the lines of "Make This Collection Public." 

- Opting to make the collection public will allow it to be shared and visible to anyone who has the link.
- Private collections will not be visible in the search results.
- Collections made public will not be visible in the search results.

**Comment by raybrarian** [comment]
FYI, when QAing a separate issue, a private collection I created is listed in the item show pages:
![image](https://user-images.githubusercontent.com/4460510/84668743-d486e280-aed8-11ea-8829-5a3a400ab6b5.png)

As far as I know, it's visible only to me, and only once I'm logged in (though we should confirm that). If that's the behavior we want, we should have any "My Collections" collections separated out/visibly distinct from the other collections in that area of the show page.

**Comment by straleyb** [comment]
Switch collection display to non_user_collections on item show page

**Comment by CGillen** [comment]
Searching on a my collection name will display all works in the collection regardless of user status

**Comment by straleyb** [comment]
Im not sure where this ticket is at or what it's testing anymore. However, functionality of the collection display works as such now:

On the item show page, the collections should now be displaying properly with their visibilities so the user knows what they can see vs what everyone else can see.

On search results, when a collection is searched for, it properly displays the right amount of works. Private works don't show up in the collection works count if you cannot see them and do show up when you can see them.

**Comment by CGillen** [comment]
#2707 is a good improvement to this ticket, but I think the main issue left is when a search is performed that **_would_** return a user collection (already blocked by a search builder), the works in that collection still return.

**Comment by shieldsb** [comment]
QA pass
